In [12]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
Metagenomic= pd.read_csv(r'C:\Users\Akintayo\Desktop\Ulcer\General Folder\Metagenomic_data.csv')

C:\Users\Akintayo\AppData\Local\Temp\ipykernel_17260\1010761004.py:1: DtypeWarning: Columns (0,1,2,3,6,10,14,26,29,30,31,37,38,39,40,41,44,45,46,47,48,49,50,55,56,57,58,59,61,63,66,67,68,72,75,79,80,81,82,83,84,87,88,89,90,91,92,98,99,100) have mixed types. Specify dtype option on import or set low_memory=False.
  Metagenomic= pd.read_csv(r'C:\Users\Akintayo\Desktop\Ulcer\General Folder\Metagenomic_data.csv')


In [3]:
Metagenomic

Superclass                                        Class  \
0               METABOLISM                  Nucleosides and Nucleotides   
1                   ENERGY  Energy and Precursor Metabolites Generation   
2               METABOLISM         Fatty Acids, Lipids, and Isoprenoids   
3       MEMBRANE TRANSPORT                           Membrane Transport   
4               METABOLISM         Fatty Acids, Lipids, and Isoprenoids   
...                    ...                                          ...   
593122                 NaN                                          NaN   
593123                 NaN                                          NaN   
593124                 NaN                                          NaN   
593125                 NaN                                          NaN   
593126                 NaN                                          NaN   

                  Subclass                                     Subsystem Name  \
0                      NaN                             Adenosyl nucleosidases   
1       Central Metabolism  Pyruvate metabolism I: anaplerotic reactions, PEP   
2              Fatty acids                               Fatty acid synthesis   
3                      NaN  Utilization systems for glycans and polysaccha...   
4              Fatty acids                    Fatty Acid Biosynthesis cluster   
...                    ...                                                ...   
593122                 NaN                                                NaN   
593123                 NaN                                                NaN   
593124                 NaN                                                NaN   
593125                 NaN                                                NaN   
593126                 NaN                                                NaN   

        Gene Count  Role Count  Active  \
0             29.0         3.0  active   
1             53.0         7.0  active   
2             78.0        10.0  active   
3            676.0         7.0  active   
4             46.0         5.0  active   
...            ...         ...     ...   
593122         NaN         NaN     NaN   
593123         NaN         NaN     NaN   
593124         NaN         NaN     NaN   
593125         NaN         NaN     NaN   
593126         NaN         NaN     NaN   

                                          Genome  Genome ID         Accession  \
0                           Bacteria SRX12677509    2.45918  2.45918.con.0001   
1                           Bacteria SRX12677509    2.45918  2.45918.con.0001   
2                           Bacteria SRX12677509    2.45918  2.45918.con.0001   
3                           Bacteria SRX12677509    2.45918  2.45918.con.0001   
4                           Bacteria SRX12677509    2.45918  2.45918.con.0001   
...                                          ...        ...               ...   
593122  Bacteria SRX12677528shotgun metagenomics    2.41875  2.41875.con.1263   
593123  Bacteria SRX12677528shotgun metagenomics    2.41875  2.41875.con.1264   
593124  Bacteria SRX12677528shotgun metagenomics    2.41875  2.41875.con.1265   
593125  Bacteria SRX12677528shotgun metagenomics    2.41875  2.41875.con.1266   
593126  Bacteria SRX12677528shotgun metagenomics    2.41875  2.41875.con.1267   

        ... E-value.1  Pathway ID.1  Pathway Name.1 Pathway Class.1  \
0       ...       NaN           NaN             NaN             NaN   
1       ...       NaN           NaN             NaN             NaN   
2       ...       NaN           NaN             NaN             NaN   
3       ...       NaN           NaN             NaN             NaN   
4       ...       NaN           NaN             NaN             NaN   
...     ...       ...           ...             ...             ...   
593122  ...       NaN           NaN             NaN             NaN   
593123  ...       NaN           NaN             NaN             NaN   
593124  ...       NaN    

In [6]:
# Display basic information about the dataset and first few rows for initial inspection
data_info = Metagenomic.info()
data_head = Metagenomic.head()
data_info, Metagenomic.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 593127 entries, 0 to 593126
Columns: 106 entries, Superclass to Gene Conservation.1
dtypes: float64(49), object(57)
memory usage: 479.7+ MB


(None,
            Superclass                                        Class  \
 0          METABOLISM                  Nucleosides and Nucleotides   
 1              ENERGY  Energy and Precursor Metabolites Generation   
 2          METABOLISM         Fatty Acids, Lipids, and Isoprenoids   
 3  MEMBRANE TRANSPORT                           Membrane Transport   
 4          METABOLISM         Fatty Acids, Lipids, and Isoprenoids   
 
              Subclass                                     Subsystem Name  \
 0                 NaN                             Adenosyl nucleosidases   
 1  Central Metabolism  Pyruvate metabolism I: anaplerotic reactions, PEP   
 2         Fatty acids                               Fatty acid synthesis   
 3                 NaN  Utilization systems for glycans and polysaccha...   
 4         Fatty acids                    Fatty Acid Biosynthesis cluster   
 
    Gene Count  Role Count  Active                Genome  Genome ID  \
 0        29.0         3.0  ac

In [7]:
# Summary of unique classes and counts within the dataset to understand predominant functional categories
class_counts = Metagenomic['Class'].value_counts()

# Examine the distribution of Superclasses as well
superclass_counts = Metagenomic['Superclass'].value_counts()

class_counts, superclass_counts


(Class
 Protein Synthesis                                               3332
 Amino Acids and Derivatives                                     2469
 Cofactors, Vitamins, Prosthetic Groups                          2442
 Energy and Precursor Metabolites Generation                     1834
 Membrane Transport                                              1777
 Stress Response, Defense and Virulence                          1760
 DNA Processing                                                  1733
 Carbohydrates                                                   1526
 Cell Cycle, Cell Division and Death                             1333
 RNA Processing                                                  1240
 Fatty Acids, Lipids, and Isoprenoids                            1035
 Nucleosides and Nucleotides                                      912
 Protein Fate (folding, modification, targeting, degradation)     796
 Respiration                                                      760
 Cell Envelop

Identify specific bacteria and genes linked to gut ulcers

In [26]:
# Filter the data for rows related to "Stress Response, Defense and Virulence" and inspect unique values in the "Role Name" and "Product" columns
stress_defense_virulence_data = Metagenomic[Metagenomic['Class'] == 'Stress Response, Defense and Virulence']

# Analyze unique Role Names and Products in this filtered data
unique_role_names = stress_defense_virulence_data['Role Name'].unique()
unique_products = stress_defense_virulence_data['Product'].unique()

unique_role_names, unique_products


(array([nan, 'tmRNA-binding protein SmpB',
        'Translation elongation factor G',
        'Translation elongation factor LepA',
        'Trk potassium uptake system protein TrkH',
        '1-deoxy-D-xylulose 5-phosphate reductoisomerase (EC 1.1.1.267)',
        'SSU ribosomal protein S10p (S20e)',
        'Oxygen-independent coproporphyrinogen-III oxidase-like protein YggW',
        'SSU ribosomal protein S1p', 'Putative pre-16S rRNA nuclease YqgF',
        'Glycine betaine ABC transport system, ATP-binding protein OpuAA (EC 3.6.3.32)',
        'Aerotolerance protein BatE',
        'Nucleoside 5-triphosphatase RdgB (dHAPTP, dITP, XTP-specific) (EC 3.6.1.66)',
        'DNA topoisomerase IV subunit A (EC 5.99.1.3)',
        'UDP-N-acetylglucosamine 1-carboxyvinyltransferase (EC 2.5.1.7)',
        'SSU ribosomal protein S12p (S23e)',
        'LSU ribosomal protein L6p (L9e)',
        'Lactoylglutathione lyase (EC 4.4.1.5)',
        'DNA topoisomerase IV subunit B (EC 5.99.1.3)',
     

In [31]:
# Extracting specific enzymes and associated bacteria names from "Product" and "Subsystem Name"
# Filtering rows where we suspect relevance to ulcer-related bacterial functions

# Keywords related to the enzymes and functions identified in the insights
enzyme_keywords = ["Tetracycline", "Macrolide", "Sphingomyelinase", "Choloylglycine hydrolase", "Catalase", 
                   "Glutathione Peroxidase", "DNA gyrase", "Topoisomerase IV", "Aerotolerance"]

# Filter rows containing any of the enzyme keywords in the Product column
ulcer_related_data = Metagenomic[Metagenomic['Product'].str.contains('|'.join(enzyme_keywords), case=False, na=False)]

# Count occurrences of unique enzymes and bacteria mentions to identify prevalence
enzyme_counts = ulcer_related_data['Product'].value_counts()

# Display the filtered data and enzyme occurrences for further insights
ulcer_related_data[['Subsystem Name', 'Product']], enzyme_counts


(       Subsystem Name                                            Product
 398               NaN                 DNA gyrase subunit B (EC 5.99.1.3)
 447               NaN       DNA topoisomerase IV subunit A (EC 5.99.1.3)
 1006              NaN                 DNA gyrase subunit A (EC 5.99.1.3)
 1012              NaN                         Aerotolerance protein BatE
 1013              NaN                         Aerotolerance protein BatD
 ...               ...                                                ...
 550339            NaN                 DNA gyrase subunit B (EC 5.99.1.3)
 550343            NaN                 DNA gyrase subunit A (EC 5.99.1.3)
 550362            NaN                 DNA gyrase subunit A (EC 5.99.1.3)
 550363            NaN  Tetracycline resistance, ribosomal protection ...
 550366            NaN  Tetracycline resistance, ribosomal protection ...
 
 [1875 rows x 2 columns],
 Product
 DNA gyrase subunit A (EC 5.99.1.3)                                        

Product: Lists the enzymes and proteins identified in the dataset that are relevant to gut health and potentially linked to ulcer conditions. Examples include: 
1. DNA gyrase subunit A/B (EC 5.99.1.3): Enzymes critical for bacterial DNA replication and repair, highly prevalent in pathogens.
2. Aerotolerance proteins (e.g., BatE, BatD): Enable bacteria to survive in variable oxygen environments.
3. Tetracycline resistance proteins (e.g., Tet(Q)): Indicate antibiotic resistance, complicating treatment of bacterial infections.
4. Sphingomyelinase (EC 3.1.4.12): Implicated in breaking down host membrane lipids, which could lead to tissue damage.
5. Catalase (KatE): Frequently found in oxidative stress resistance (16 occurrences). This enzyme allows bacteria to detoxify reactive oxygen species, enabling survival in the gut.


High Counts of Resistance Genes: Antibiotic resistance genes like Tetracycline resistance (Tet(Q)) and Macrolide resistance (Mef(A)) suggest the gut microbiota may harbor resistant pathogens, potentially complicating treatments for infections.


Pathogenicity Indicators:
Sphingomyelinase and Choloylglycine hydrolase are associated with host tissue damage and bile acid metabolism, processes that can irritate the gut lining and exacerbate ulcers.


Aerotolerance proteins (BatA-E) are found frequently, pointing to adaptive bacteria capable of surviving in both oxygen-rich and anaerobic gut regions, some of which might be opportunistic pathogens.



To analyze potential links to gut ulcers in humans:

Investigate rows mentioning gastrointestinal or pathogenic bacteria often associated with ulcers, such as Helicobacter pylori.

Review pathways and roles tied to mechanisms of ulcer formation, like ammonia production, urease activity, and cytotoxin production.


Highlight pathways or genes that could influence inflammation or ulcer development in the gut.

In [41]:
# Filter data for entries that may include bacteria associated with gut ulcers (e.g., Helicobacter pylori).
# Checking if there's any mention of "Helicobacter" or similar keywords in the 'Genome Name' or 'Source Organism' columns.
ulcer_related_data = Metagenomic[
    Metagenomic['Genome Name'].str.contains("Helicobacter", case=False, na=False) |
    Metagenomic['Source Organism'].str.contains("Helicobacter", case=False, na=False)
]

# Display the filtered data relevant to ulcer-causing bacteria.
ulcer_related_data


Superclass                                        Class  \
32856      ENERGY  Energy and Precursor Metabolites Generation   
33124         NaN                                          NaN   
349731     ENERGY  Energy and Precursor Metabolites Generation   
350155        NaN                                          NaN   

                  Subclass          Subsystem Name  Gene Count  Role Count  \
32856         Fermentation  Fermentations: Lactate        34.0         7.0   
33124                  NaN                     NaN         NaN         NaN   
349731  Central Metabolism           Citrate lyase        17.0         6.0   
350155                 NaN                     NaN         NaN         NaN   

        Active                Genome  Genome ID         Accession  ...  \
32856   active  Bacteria SRX12677516    2.45914  2.45914.con.0001  ...   
33124      NaN  Bacteria SRX12677516    2.45914  2.45914.con.0002  ...   
349731  active  Bacteria SRX12677518    2.45923  2.45923.con.0001  ...   
350155     NaN  Bacteria SRX12677518    2.45923  2.45923.con.0002  ...   

       E-value.1  Pathway ID.1  Pathway Name.1 Pathway Class.1 Annotation.3  \
32856        NaN           NaN             NaN             NaN          NaN   
33124        NaN           NaN             NaN             NaN          NaN   
349731       NaN           NaN             NaN             NaN          NaN   
350155       NaN           NaN             NaN             NaN          NaN   

       Unique Genome Count.1  Unique Gene Count.1  Unique EC Count.1  \
32856                    NaN                  NaN                NaN   
33124                    NaN                  NaN                NaN   
349731                   NaN                  NaN                NaN   
350155                   NaN                  NaN                NaN   

        EC Conservation.1 Gene Conservation.1  
32856                 NaN                 NaN  
33124                 NaN                 NaN  
349731                NaN                 NaN  
350155                NaN                 NaN  

[4 rows x 106 columns]

In [42]:
ulcer_related_data.to_csv("Bacteria_associated_Ulcer.csv")

Analyze potential indirect mechanisms related to gut ulcer development, Focus on data entries that may play a role in:

Toxin Production: Certain bacterial toxins can damage the gastric lining.

Inflammatory Response: Bacterial products that activate immune responses may contribute to chronic inflammation, leading to ulceration.

Urease Activity and pH Modulation: Some bacteria produce urease, which increases local pH and could disrupt the gut environment.

Amino Acid Metabolism: Specific amino acid pathways, particularly involving lysine or arginine, can influence bacterial growth and immune evasion.

In [52]:
# Filtering the data for entries related to toxins, inflammatory response, urease activity, and amino acid metabolism.
# Searching relevant keywords in 'Subsystem Name', 'Role Name', 'Function', and 'Pathway Name' to capture these areas.

keywords = ["toxin", "inflammation", "urease", "amino acid", "pH", "immune"]

# Filter the dataset for any rows containing these keywords in specific columns
indirect_mechanism_data = Metagenomic[
    Metagenomic['Subsystem Name'].str.contains('|'.join(keywords), case=False, na=False) |
    Metagenomic['Role Name'].str.contains('|'.join(keywords), case=False, na=False) |
    Metagenomic['Function'].str.contains('|'.join(keywords), case=False, na=False) |
    Metagenomic['Pathway Name'].str.contains('|'.join(keywords), case=False, na=False)
]

# Display the filtered data relevant to indirect mechanisms in ulcer development.
indirect_data_ulcer= indirect_mechanism_data[['Subsystem Name', 'Role Name', 'Function', 'Pathway Name']]

indirect_data_ulcer


Subsystem Name Role Name  \
4                         Fatty Acid Biosynthesis cluster       NaN   
11      Nudix proteins (nucleoside triphosphate hydrol...       NaN   
12             Translation termination factors, bacterial       NaN   
14                                    Glycogen metabolism       NaN   
16                           De Novo Pyrimidine Synthesis       NaN   
...                                                   ...       ...   
550347                                                NaN       NaN   
550349                                                NaN       NaN   
550353                                                NaN       NaN   
550359                                                NaN       NaN   
550372                                                NaN       NaN   

                                                 Function  \
4       1-deoxy-D-xylulose 5-phosphate reductoisomeras...   
11                 Dihydropteroate synthase (EC 2.5.1.15)   
12                                                    NaN   
14      3-oxoacyl-[acyl-carrier-protein] synthase, KAS...   
16      1-deoxy-D-xylulose 5-phosphate reductoisomeras...   
...                                                   ...   
550347  Streptozocin;[[(3R,4R,5S,6R)-3-(butanoylamino)...   
550349  CDP-diacylglycerol--glycerol-3-phosphate 3-pho...   
550353  Glycerophosphoryl diester phosphodiesterase (E...   
550359  Glycerophosphoryl diester phosphodiesterase (E...   
550372  Glycerophosphoryl diester phosphodiesterase (E...   

                                             Pathway Name  
4                                 Fatty acid biosynthesis  
11          Glycosphingolipid biosynthesis - globo series  
12        Glycosphingolipid biosynthesis - ganglio series  
14      Phenylalanine, tyrosine and tryptophan biosynt...  
16                              Tetracycline biosynthesis  
...                                                   ...  
550347                                                NaN  
550349                                                NaN  
550353                                                NaN  
550359                                                NaN  
550372                                                NaN  

[5765 rows x 4 columns]

Identifying Bacteria and Gene Functions: search for bacteria or genes in the dataset that might be tied to ulcer pathogenesis and gut health, looking for known gut-associated bacterial groups.


Pathway and Function Insights: provide an analysis of pathways and functions that contribute to ulcer pathogenesis based on the filtered data, especially pathways involved in inflammation, toxin production, and cellular adhesion.


Enzymes and Associations: By reviewing specific enzymes related to these functions, we can gain insights into their role in gut health and their connection to bacterial species known to influence ulcer formation.


Special Proteins and Functional Insights: I will assess any proteins that may play a role in gut health, especially those involved in bacterial survival in acidic environments or interactions with host tissues.


Pathogenesis and Species Identification: Based on the roles, enzymes, and pathways identified, I will aim to link any relevant bacteria or bacterial species to ulcer pathogenesis.

In [35]:
# Identify gut-related bacteria or gene functions
# Define keywords to filter gut-related terms
gut_related_keywords = ["pathogenesis", "virulence", "adhesion", "gut", "ulcer", "cytotoxin", "immune", "acid resistance"]

# Filter dataset for relevant keywords in columns related to functions and pathways
gut_bacteria_data = Metagenomic[
    Metagenomic['Role Name'].str.contains('|'.join(gut_related_keywords), case=False, na=False) |
    Metagenomic['Function'].str.contains('|'.join(gut_related_keywords), case=False, na=False) |
    Metagenomic['Pathway Name'].str.contains('|'.join(gut_related_keywords), case=False, na=False)
]

# Display results for gut-related bacteria or gene functions
print("Gut-related bacteria or gene functions:")
print(gut_bacteria_data[['Source Organism', 'Role Name', 'Function', 'Pathway Name']])

# Analyze pathways and functional insights
# show pathways related to inflammation, acid resistance, or cellular adhesion
pathway_keywords = ["inflammation", "acid resistance", "adhesion", "toxin"]
pathway_insights = Metagenomic[
    Metagenomic['Pathway Name'].str.contains('|'.join(pathway_keywords), case=False, na=False)
]

# Display pathway insights
print("\nPathways and Functional Insights:")
print(pathway_insights[['Pathway Name', 'Pathway Class', 'Function']])

# Investigate specific enzymes and associations with bacterial groups
# Show enzymes that might be relevant to gut health and ulcer pathogenesis
enzyme_keywords = ["urease", "protease", "cytotoxin", "lipopolysaccharide", "LPS"]
enzyme_data = Metagenomic[
    Metagenomic['Role Name'].str.contains('|'.join(enzyme_keywords), case=False, na=False) |
    Metagenomic['Product'].str.contains('|'.join(enzyme_keywords), case=False, na=False)
]

# Display specific enzymes related to gut health
print("\nSpecific Enzymes and Associations with Gut Health:")
print(enzyme_data[['Source Organism', 'Role Name', 'Product', 'Function']])

# Explore special proteins for gut-related bacteria
# Identify proteins that may influence gut health (e.g., adhesins, invasins)
protein_keywords = ["adhesion", "invasion", "toxins", "LPS"]
protein_data = Metagenomic[
    Metagenomic['Role Name'].str.contains('|'.join(protein_keywords), case=False, na=False) |
    Metagenomic['Product'].str.contains('|'.join(protein_keywords), case=False, na=False)
]

# Display special proteins relevant to gut health
print("\nSpecial Proteins Relevant to Gut Health:")
print(protein_data[['Source Organism', 'Role Name', 'Product', 'Function']])

# Identify species and pathogenic mechanisms
# Use the filtered data to link source organisms with gut ulcer-related mechanisms
species_mechanisms = gut_bacteria_data[['Source Organism', 'Role Name', 'Pathway Name', 'Function']]

# Display species and their related pathogenic mechanisms
print("\nSpecies and Their Pathogenic Mechanisms:")
print(species_mechanisms)


Gut-related bacteria or gene functions:
                                 Source Organism Role Name  \
349888  Bifidobacterium longum (strain NCC 2705)       NaN   
475456                    Salmonella typhimurium       NaN   
475571                    Salmonella typhimurium       NaN   
475678                    Salmonella typhimurium       NaN   
475869                    Salmonella typhimurium       NaN   
476077                    Salmonella typhimurium       NaN   
476547                          Escherichia coli       NaN   
476584                          Escherichia coli       NaN   
476764                          Escherichia coli       NaN   
477376                          Escherichia coli       NaN   

                                                 Function Pathway Name  
349888  the acid resistance membrane protein (hded) fa...          NaN  
475456  the type iii (virulence-related) secretory pat...          NaN  
475571  the type iii (virulence-related) secretory pat... 

Key Organisms Identified:

1. Bifidobacterium longum (strain NCC 2705): A beneficial gut bacterium known for promoting gut health.
2. Salmonella typhimurium: A pathogenic bacterium associated with gastrointestinal infections.
3. Escherichia coli: Includes both commensal and pathogenic strains, some of which are involved in gut-related diseases.
4. Bacteroides fragilis: A common gut bacterium with roles in digestion and immune modulation but also capable of causing infections under certain conditions.



Relevant Gene Functions and Pathways
Acid Resistance: Bifidobacterium longum and Escherichia coli feature acid resistance mechanisms, such as membrane proteins (e.g., HdeD) that help bacteria survive in the acidic gut environment.

Virulence Mechanisms: Salmonella typhimurium employs Type III secretion systems, a hallmark of virulence that allows bacteria to inject proteins into host cells, disrupting normal functions and causing inflammation.

Metabolism and Transport: Some bacteria exhibit roles in glucitol metabolism (E. coli), highlighting their adaptation to diverse nutrient sources in the gut.



Special Proteins and Products
1. Lipopolysaccharide (LPS) biosynthesis and assembly proteins (e.g., LptA, LptD): Identified in species like Bacteroides fragilis. LPS is a critical component of the bacterial outer membrane, playing roles in immune evasion and inflammation.

2. Zinc protease and ADP-heptose:LPS heptosyltransferase: Enzymes involved in bacterial survival and pathogenicity, found in Bacteroides fragilis.



Potential Implications for Gut Ulcers

Pathogens: The presence of virulent strains of Salmonella typhimurium and pathogenic E. coli suggests potential involvement in gut inflammation and damage, precursors to ulcers.

Acid Resistance Mechanisms: Identified in E. coli and Bifidobacterium longum, these traits are critical for bacteria surviving in the stomach and upper GI tract, regions susceptible to ulceration.

LPS and Immune Response: LPS from gram-negative bacteria can trigger strong immune responses, potentially leading to chronic inflammation and ulcer formation.

Identify Gut Related Bacteria through pathway

In [38]:
# Filter for entries that mention LPS or protease in the Role Name or Function columns
lps_protease_data = Metagenomic[
    Metagenomic['Role Name'].str.contains("LPS|protease", case=False, na=False) |
    Metagenomic['Function'].str.contains("LPS|protease", case=False, na=False)
]

# Display LPS and protease-related entries
print("Functional Insights from LPS and Proteases:")
print(lps_protease_data[['Source Organism', 'Role Name', 'Function', 'Pathway Name']])



Functional Insights from LPS and Proteases:
                       Source Organism  \
33152    Megamonas hypermegale ART12/1   
98581   Streptococcus pneumoniae TIGR4   
98592                              NaN   
98669                    Streptococcus   
98901                              NaN   
...                                ...   
536230                             NaN   
536309                             NaN   
536614                             NaN   
536806                             NaN   
536904                             NaN   

                                                Role Name  \
33152                                                 NaN   
98581   ATP-dependent Clp protease proteolytic subunit...   
98592   Cell division-associated, ATP-dependent zinc m...   
98669                           LPS-assembly protein LptD   
98901                            FIG015287: Zinc protease   
...                                                   ...   
536230  ATP-dependent Cl

In [39]:
# Extract unique pathways related to LPS and proteases
lps_protease_pathways = lps_protease_data[['Pathway Name', 'Pathway Class', 'Function']].drop_duplicates()

# Display pathways associated with LPS and proteases
print("Pathways Associated with LPS and Proteases:")
print(lps_protease_pathways)


Pathways Associated with LPS and Proteases:
                                    Pathway Name  \
33152                                        NaN   
98581                    Sphingolipid metabolism   
98592            Arginine and proline metabolism   
98669                    Caprolactam degradation   
98901                                        NaN   
98929                                        NaN   
98944                                        NaN   
99011                                        NaN   
475494                                       NaN   
475621                                       NaN   
529318                 One carbon pool by folate   
529319                      Histidine metabolism   
529341  Glycine, serine and threonine metabolism   

                                    Pathway Class  \
33152                                         NaN   
98581                            Lipid Metabolism   
98592                       Amino Acid Metabolism   
98669   Xenobio

In [40]:
# Hypothetical mapping of enzymes/proteins to common gut pathogens
hypothetical_pathogens = {
    'Escherichia coli': ['LPS', 'protease'],
    'Salmonella': ['LPS', 'protease'],
    'Helicobacter pylori': ['urease', 'protease'],
    'Campylobacter jejuni': ['cytotoxin', 'LPS']
}

# Display mapping for reference
print("Hypothetical Gut Pathogens and Associated Enzymes/Proteins:")
for pathogen, functions in hypothetical_pathogens.items():
    print(f"{pathogen}: {', '.join(functions)}")


Hypothetical Gut Pathogens and Associated Enzymes/Proteins:
Escherichia coli: LPS, protease
Salmonella: LPS, protease
Helicobacter pylori: urease, protease
Campylobacter jejuni: cytotoxin, LPS


Summary of Pathogenic Mechanisms
Pathogen	Key Enzymes/Proteins	Mechanism of Action	Role in Ulcers
Escherichia coli	LPS, protease	Triggers inflammation and disrupts host proteins	Can lead to mucosal damage and inflammation, precursors to ulcers
Salmonella	LPS, protease	Promotes invasion, inflammation, and immune evasion	Chronic infections may impair the mucosal layer, predisposing to ulcer formation
Helicobacter pylori	Urease, protease	Neutralizes acid, damages mucus and epithelial cells	Directly causes gastric and duodenal ulcers by disrupting protective barriers
Campylobacter jejuni	Cytotoxin, LPS	Induces cell death and inflammation	Severe infections can cause intestinal lesions akin to ulcers